app/src/main/java/org/mifos/mobile/ui/loan_account_transaction/LoanAccountTransactionFragment.kt

```diff
@@ -0,0 +1,63 @@
+package org.mifos.mobile.ui.loan_account_transaction
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.fragment.app.viewModels
+import dagger.hilt.android.AndroidEntryPoint
+import org.mifos.mobile.core.ui.theme.MifosMobileTheme
+import org.mifos.mobile.ui.activities.base.BaseActivity
+import org.mifos.mobile.ui.fragments.base.BaseFragment
+import org.mifos.mobile.utils.Constants
+
+/*
+~This project is licensed under the open source MPL V2.
+~See https://github.com/openMF/self-service-app/blob/master/LICENSE.md
+*/ /**
+ * Created by dilpreet on 4/3/17.
+ */
+@AndroidEntryPoint
+class LoanAccountTransactionFragment : BaseFragment() {
+
+    private val viewModel: LoanAccountTransactionViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (activity as? BaseActivity)?.showToolbar()
+        if (arguments != null) {
+            viewModel.setLoanId(arguments?.getLong(Constants.LOAN_ID))
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        (activity as BaseActivity).hideToolbar()
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                MifosMobileTheme {
+                    LoanAccountTransactionScreen(
+                        navigateBack = { activity?.supportFragmentManager?.popBackStack() }
```

Comment:
can it be replaced by `onBackPressedDispatcher.onBackPressed()` ?

Reply:
`activity?.supportFragmentManager?.popBackStack()`: Use this to programmatically navigate back (pop) through fragment transactions and manage the fragment back stack within an activity or fragment.

`onBackPressedDispatcher.onBackPressed()`: Use this to programmatically simulate the default back button behavior (like pressing the device's back button) from contexts where direct access to onBackPressed() of the activity is not available or convenient (e.g., within ViewModels, custom classes, etc.).

So, by these definitions, I believe we should go with pop backstack()? What are your views?